src/features/profile/api/useUser.ts

```diff
@@ -1,27 +1,15 @@
-import { useCallback, useEffect } from 'react'
+import { useCallback } from 'react'
 import Router from 'next/router'
 import { trpc } from '~/utils/trpc'
 
 export const useUser = ({ redirectTo = '', redirectIfFound = false } = {}) => {
-  const { data: user, isFetching, ...rest } = trpc.me.get.useQuery()
-
-  useEffect(() => {
-    // if no redirect needed, just return (example: already on /dashboard)
-    // if user data not yet there (fetch in progress, logged in or not) then don't do anything yet
-    if (!redirectTo || isFetching) return
-
-    const shouldRedirect =
-      // If redirectTo is set,
-      redirectTo &&
-      // redirect if the user was not found.
-      ((!redirectIfFound && !user) ||
-        // If redirectIfFound is also set, redirect if the user was found
-        (redirectIfFound && user))
-
-    if (shouldRedirect) {
-      Router.push(redirectTo)
-    }
-  }, [user, redirectIfFound, redirectTo, isFetching])
+  const [user] = trpc.me.get.useSuspenseQuery(undefined, {
+    onSuccess: () => {
```

Comment:
comment: do we no longer redirect if the user is unauthorised? seems like nowhere else in the code does the auth gate happen

Reply:
We do, this redirected is done on the top level `ErrorBoundary` instead. I've wrapped the app root with an `ErrorBoundary`, and all `unauthorized` TRPCErrors will do a manual relocation. Code block below.

```     
 if (res.data === 'UNAUTHORIZED') {
        const params = new URLSearchParams(window.location.search)

        const callbackUrl = params.get('callbackUrl')

        window.location.href = !!callbackUrl
          ? `sign-in/?${CALLBACK_URL_KEY}=${callbackUrl}`
          : `sign-in`

        return
      }
```